### PR TITLE
feat: adding show command, to show details of listener

### LINKED ISSUE #4419: Create clap command to show the details of an existing TCP listener on a node
### Current behavior

Missing command for `tcp-listener show`.

### Desired behavior

Implement a command that retrieves from a running node the information of a `tcp-listener` given its ID.

```bash
ockam tcp-listener show [ID] --at [node-name]
```

Required args:
- ID

Optional args:
- at: defaults to the default node

Output:

```bash
TCP Listener:
  ID: {}
  Type: {}
  Mode: {}
  Socket address: {}
  Worker address: {}
```

Similar command: [`tcp-listener delete`](https://github.com/build-trust/ockam/blob/5837c73b90cfa72a76879642e727024c9e0977d0/implementations/rust/ockam/ockam_command/src/tcp/listener/delete.rs#L13)

### PR DESCRIPTION
**Pull request by 0xkelvin**
## Checks

- [x] All commits in this Pull Request are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) and Verified by Github.
- [x] All commits in this Pull Request follow the Ockam [commit message convention](https://github.com/build-trust/.github/blob/main/CONTRIBUTING.md#commit-messages).
- [x] I accept the Ockam Community [Code of Conduct](https://github.com/build-trust/.github/blob/main/CODE_OF_CONDUCT.md).
- [x] I have accepted the Ockam [Contributor License Agreement](https://github.com/build-trust/ockam-contributors/blob/main/CLA.md) by adding my Git/Github details in a row at the end of the [CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/blob/main/CONTRIBUTORS.csv) file in a separate pull request to the [build-trust/ockam-contributors](https://github.com/build-trust/ockam-contributors) repository. The easiest way to do this is to [edit the CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/edit/main/CONTRIBUTORS.csv) file in the github web UI and create a PR, this will mark the commit as verified.

### CODE REVIEW

**Comment by 0xkelvin** [comment]
this PR is the implementation for this https://github.com/build-trust/ockam/issues/4419

**Comment by adrianbenavides** [comment]
> this PR is the implementation for this #4419

A small tip: if you write "Fix {url of the issue}" in the PR description it gets automatically linked (you will see it in the `Development` section on the right sidebar.

File: implementations/rust/ockam/ockam_command/src/tcp/listener/show.rs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    }
+}
+
+async fn run_impl(
+    ctx: Context,
+    (opts, cmd): (CommandGlobalOpts, ShowCommand),
+) -> crate::Result<()> {
+    let node = extract_address_value(&cmd.node_opts.at)?;
+    let mut rpc = Rpc::background(&ctx, &opts, &node)?;
+    rpc.request(Request::get("/node/tcp/listener")).await?;
```
Comment (adrianbenavides) [inline implementations/rust/ockam/ockam_command/src/tcp/listener/show.rs]: You have to add a new endpoint to the NodeManager to respond to this new request. Note that the endpoint must include the `id` of the tcp-listener you want to retrieve.

Check this PR as an example: https://github.com/build-trust/ockam/pull/4446

File: implementations/rust/ockam/ockam_command/src/tcp/listener/show.rs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    /// Tcp Listener ID
+    pub id: String,
+}
+
+#[derive(Clone, Debug, Args)]
+pub struct TCPlistenerNodeOpts {
+    /// Node at which to create the listener
+    #[arg(global = true, long, value_name = "NODE", default_value_t = default_node_name())]
+    pub at: String,
+}
```
Comment (adrianbenavides) [inline implementations/rust/ockam/ockam_command/src/tcp/listener/show.rs]: You can simplify this and inline the `at` argument within the `ShowCommand` struct.

**Review by adrianbenavides — CHANGES_REQUESTED** [review]
Thanks @0xkelvin! It looks great so far, but there a couple of parts missing:
- Implement the NodeManager endpoint.
- Add a bats test

**Review by adrianbenavides — APPROVED** [review]

**Pull request merged**